components/places/src/storage/history.rs

```diff
@@ -1177,6 +1180,29 @@ pub fn get_visited_urls(
     )?)
 }
 
+pub fn get_top_frecent_site_infos(
+    db: &PlacesDb,
+    num_items: i32,
+    top_site_frecency: i32,
+) -> Result<TopFrecentSiteInfos> {
+    let infos = db.query_rows_and_then_named_cached(
+        "SELECT frecency, title, url
+         FROM moz_places
+         WHERE frecency >= :frecencyThreshold
+           AND (SUBSTR(url, 1, 6) == 'https:' OR SUBSTR(url, 1, 5) == 'http:')
+           AND (last_visit_date_local + last_visit_date_remote) != 0 AND
+           NOT hidden
+         ORDER BY frecency DESC
+         LIMIT :limit",
+        rusqlite::named_params! {
+            ":limit": num_items,
+            ":frecencyThreshold": top_site_frecency
+        },
+        TopFrecentSiteInfo::from_row,
+    )?;
+    Ok(TopFrecentSiteInfos { infos })
```

Comment:
Do we need to do any collapsing or deduping here, or would you like to save that for a follow-up? I'm fine with getting this landed as-is so you can build out the a-c side, and copying more of Desktop later if we decide we need it.

Reply:
Gonna think about this as a follow up since we can do that either here or in A-C.